### DISCUSSION TITLE
temporary and overlapping centroids

### DISCUSSION THREAD

**Discussion by EJMUL** [bmcclintock/momentuHMM#77] [Q&A] [answered]
Is there a possibility in momentuHMM to use temporary centers of attraction? For example, I have a dataset of killer whale telemetry data, and fishery catch locations. The fishery lasts for a few hours (remains relatively stationary) and then stops. The boat leaves, but I have no AIS data from the vessels, only the locations of where they were fishing. I would like to investigate if this attracts the killer whales by using a 3 state HMM (foraging, transit, attracted to fishery). The fishing events are only temporary, but several may occur at the same time, in which case the whales may be attracted to one of them. Is there a way to model this in momentuHMM? I think I need to use the centroids option, but in predata, I get the error message "centroids time does not span data; each observation time must have a corresponding entry in centroids".

**Comment by bmcclintock** [comment, marked as answer]
I believe this can be done, but it would be complicated. 

First, one needs to be able to specify centroids, and I believe you're getting that error because each observation in the data must have a corresponding entry for the location of the centroid(s). In your case, when there are no "fishing events" a dummy entry would still need to be entered. 

That's the relatively easy part -- the challenging bit is specifying the model so that individuals can only switch to (and remain in) the "fishing" states during fishing events. In essence, covariates would be needed for the state transition probabilities that indicate when each centroid is "active", and these covariates could then be used to force transitions to the centroid states to be impossible whenever they are not active. And when a centroid is "active", you would likely need to make the strength of attraction to the centroid a function of distance or other explanatory covariates (as is done in the grey seal example of the vignette).

This would be fairly straightforward with a single centroid (i.e. with only one fishing event occurring at any given time), but with multiple centroids it could likely get very complicated. My sense is that determining how to go about this for your case will require some time and a white board. Feel free to contact me about this outside of GitHub.

**Comment by bmcclintock** [comment]
Because the fishing events are relatively stationary, an alternative (and perhaps simpler) avenue to accomplish this is potential functions (e.g. [Brillinger et al. 2012](https://link.springer.com/chapter/10.1007/978-1-4614-1344-8_22)). See e.g. the "African buffalo recharge dynamics" and "Simulating constrained movement" sections of the vignette. The fishing event covariate raster for the potential function would change through time (e.g. daily) based on when and where any fishing events are, and a single state transition probability covariate would indicate time steps when any fishing event is active (and be used to force transitions to the "fishing" state to be impossible whenever there are no fishing events). Thus it would eliminate the need to account for fishing events as distinct centroids.

**Comment by EJMUL** [comment]
Dear Dr. McClintock,
Thank you for your quick replies, I appreciate it!
The second option seems a bit easier, and I think it should be possible to capture the fishing activity in a series of rasters. Eventually I hope to find some idea of what the influence is of distance to a fishery for the transition probability to change to "attracted to fishery". In addition, I am hoping to find out what the effect of time (relative to the start of the fisheries) is on the transition probability; are the whales switching to "attracted to fishery" at the start of the fishery, or a few hours later, etc. This weekend I will dive into the vignette on the African buffalo to see if I can set up a model. I would like to send you an email early next week with some further questions, and perhaps a snapshot of the data, if that is OK with you? 
Kind regards,

Evert Mul

**Comment by bmcclintock** [comment]
Sure
